ilc/client/TransitionManager/ScrollController/ScrollController.js

```diff
@@ -1,10 +1,18 @@
-export class UrlHashController {
+export class ScrollController {
     // @todo write e2e tests for hash position restoring
     #hashStoreNode = document.body;
     #hashStoreAttribute = 'ilcTempStoredHash';
+    #lastVisitedUrl = this.location.pathname;
+    #shouldScrollToTop = true;
+
+    get location() {
+        return window.location;
```

Comment:
maybe it is better to use router. I would avoid calling `window.location` because scroll process happens during page transition by triggering route.change.

Reply:
yeah, I also didn't like it, it was spread over the class. I though about injecting something upfront, but most likely it would  increase PR changes, so I decided to keep improvement small and isolate window usage within one getter. Next step might be injection from the above